### PR TITLE
Rename package shared to pkg

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: pkg/testgrid/testgrid.go [hunk 1]
```diff
@@ -24,9 +24,9 @@ import (
 	"os"
 	"path"
 
-	"knative.dev/test-infra/shared/common"
-	"knative.dev/test-infra/shared/junit"
-	"knative.dev/test-infra/shared/prow"
+	"knative.dev/test-infra/pkg/common"
+    "knative.dev/test-infra/pkg/junit"
```
Comment (mattmoor) [inline pkg/testgrid/testgrid.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/junit"
```

File: pkg/testgrid/testgrid_test.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ import (
 	"path"
 	"testing"
 
-	"knative.dev/test-infra/shared/junit"
-	"knative.dev/test-infra/shared/prow"
+    "knative.dev/test-infra/pkg/junit"
+    "knative.dev/test-infra/pkg/prow"
```
Comment (mattmoor) [inline pkg/testgrid/testgrid_test.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/junit"
	"knative.dev/test-infra/pkg/prow"
```

File: tools/prow-auto-bumper/getversion_test.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ import (
 	"knative.dev/pkg/test/ghutil"
 	"knative.dev/pkg/test/ghutil/fakeghutil"
 
-	"knative.dev/test-infra/shared/git"
+    "knative.dev/test-infra/pkg/git"
```
Comment (mattmoor) [inline tools/prow-auto-bumper/getversion_test.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/git"
```

File: images/backups/Makefile [hunk 4]
```diff
@@ -13,4 +13,4 @@
 # limitations under the License.
 
 IMAGE_NAME = backups
-include ../../shared/Makefile.simple-image
+include ../../pkg/Makefile.simple-image
```
Comment (chaodaiG) [inline images/backups/Makefile]: should this file be in pkg?

Comment (chizhg) [reply images/backups/Makefile]: I can move it to `images/`, do you think it would be a better place?

Comment (chaodaiG) [reply images/backups/Makefile]: I think that's more reasonable than in pkg

Comment (chizhg) [reply images/backups/Makefile]: Agree, moved.

**Comment by chaodaiG** [comment]
/lgtm
thank you

**Pull request merged**